FhirToDataLake/Synapse-search.yml

```diff
@@ -0,0 +1,210 @@
+      $creds = New-Object System.Management.Automation.PSCredential ($env:SQL_USERNAME, $securedPassword)
+      New-AzSynapseWorkspace -ResourceGroupName $(resourceGroup) -Name $(synapseWorkspaceName) -Location eastus2 -SqlAdministratorLoginCredential $creds `
+        -DefaultDataLakeStorageAccountName $(pipelineStorage) -DefaultDataLakeStorageFilesystem $(pipelineStorageFileSystem) -ErrorAction Stop
+
+      # Creating Synapse via Azure portal will default allow all IP to access it, while creating via Powershell will not
+      # Here manually set it firewall rules, see https://docs.microsoft.com/en-us/azure/synapse-analytics/quickstart-create-workspace-powershell
+      $FirewallParams = @{
+        WorkspaceName = '$(synapseWorkspaceName)'
+        Name = 'allowAll'
+        ResourceGroupName = '$(resourceGroup)'
+        StartIpAddress = '0.0.0.0'
+        EndIpAddress = '255.255.255.255'
+      }
+      New-AzSynapseFirewallRule @FirewallParams
+  env:
+    SQL_USERNAME: $(SqlUsername)
+    SQL_PASSWORD: $(SqlPassword)
+
+- task: AzurePowerShell@5
+  displayName: Deploy Fhir Synapse sync pipeline
+  inputs: 
+    azureSubscription: 'ResoluteOpenSource'
+    azurePowerShellVersion: latestVersion
+    ScriptType: inlineScript
+    Inline: |
+      $templateParameters = @{
+          appName = "$(appName)"
+          fhirServerUrl = "$(fhirServerUrl)"
+          PackageUrl = "$(packageUrl)"
+      }
+
+      # Function will be triggered at second 0 of every 5th minute of every hour of each day (0, 5, 10,...,55)
+      # Deploy pipeline at (0, 5, 10,...,55) so that after pipeline been created, it will not be triggered immediately
+      $secondsDelay = 5*60 - ([int](Get-Date -Format ss) + 60*([int](Get-Date -Format mm)%5))
+      Start-Sleep -s $secondsDelay
+
+      New-AzResourceGroupDeployment -Name $(deploymentName) -ResourceGroupName $(resourceGroup) -TemplateFile $(System.DefaultWorkingDirectory)/FhirToDataLake/deploy/templates/FhirSynapsePipelineTemplate.json -TemplateParameterObject $templateParameters -Verbose
+      $storageAccountName = (Get-AzResource -ResourceGroupName $(resourceGroup) -Name $(appName)* -ResourceType Microsoft.Storage/storageAccounts).Name
+      
+      # Get storage name and inject it into pipeline variable
+      Write-Host "##vso[task.setvariable variable=storageAccountName;]$storageAccountName"
+
+- task: AzurePowerShell@5
+  displayName: Fetch data from FHIR server
+  inputs: 
+    azureSubscription: 'ResoluteOpenSource'
+    azurePowerShellVersion: latestVersion
+    ScriptType: inlineScript
+    Inline: |
+      # Here delay some time so that function be triggered twice
+      Write-Host 'Delay 7 minutes to fetch data from FHIR server'
+      Start-Sleep -s 420
+
+      Write-Host 'Stop function at $(Get-Date -Format hh:MM:ss)'
+      Stop-AzFunctionApp -Name $(appName) -ResourceGroupName $(resourceGroup) -Force
+  
+- task: AzureCLI@2
+  displayName: Assign access premission roles
```

Comment:
nit, can be simplified to `Assign access roles`?

Reply:
Sure, updated